packages/components/src/Datalist/Datalist.component.test.js

```diff
@@ -0,0 +1,349 @@
+		wrapper
+			.find('input')
+			.at(0)
+			.simulate('focus');
+
+		// then
+		expect(wrapper.find(Typeahead).props().items).toEqual(['foo', 'bar', 'foobar', 'lol']);
+		expect(wrapper.find(Typeahead).props().value).toBe('foo');
+	});
+
+	it('should reset previous value on ESC keydown', () => {
+		// given
+		const wrapper = mount(
+			<Datalist
+				id={'my-datalist'}
+				isValid
+				multiSection={false}
+				errorMessage={'This should be correct'}
+				onChange={jest.fn()}
+				onFinish={jest.fn()}
+				{...props}
+				value={'foo'}
+			/>,
+		);
+		expect(wrapper.find(Typeahead).props().value).toBe('foo');
+		wrapper
+			.find('input')
+			.at(0)
+			.simulate('change', { target: { value: 'newValue' } });
+		expect(wrapper.find(Typeahead).props().value).toBe('newValue');
+
+		// when
+		wrapper
+			.find('input')
+			.at(0)
+			.simulate('keydown', { which: keycode.codes.esc });
+
+		// then
+		expect(wrapper.find(Typeahead).props().value).toBe('foo');
+	});
+
+	it('should display all suggestions on DOWN keydown when there are no suggestions yet', () => {
+		// given
+		const wrapper = mount(
+			<Datalist
+				id={'my-datalist'}
+				isValid
+				multiSection={false}
+				errorMessage={'This should be correct'}
+				onChange={jest.fn()}
+				onFinish={jest.fn()}
+				{...props}
+				value={'foo'}
+			/>,
+		);
+		expect(wrapper.find(Typeahead).props().items).toBe(null);
+
+		// when
+		wrapper
+			.find('input')
+			.at(0)
+			.simulate('keydown', { which: keycode.codes.down });
+
+		// then
+		expect(wrapper.find(Typeahead).props().items.length).toBe(4);
+	});
+
+	it('should change value on ENTER keydown with no selected suggestion', () => {
+		// given
+		const onChange = jest.fn();
+		const onFinish = jest.fn();
+		const wrapper = mount(
+			<Datalist
+				id={'my-datalist'}
+				isValid
+				multiSection={false}
+				errorMessage={'This should be correct'}
+				onChange={onChange}
+				onFinish={onFinish}
+				{...props}
+				value={'foo'}
+			/>,
+		);
+		const input = wrapper.find('input').at(0);
+		input.simulate('change', { target: { value: 'fo' } });
+		expect(onChange).not.toBeCalled();
+
+		// when
+		input.simulate('keydown', { which: keycode.codes.enter });
+
+		// then
+		const payload = { value: 'fo' };
+		expect(onChange).toBeCalledWith(expect.anything(), payload);
+		expect(onFinish).toBeCalledWith(expect.anything(), payload);
+	});
+
+	it('should reset suggestions on ENTER keydown', () => {
+		// given
+		const wrapper = mount(
+			<Datalist
+				id={'my-datalist'}
+				isValid
+				multiSection={false}
+				errorMessage={'This should be correct'}
+				onChange={jest.fn()}
+				onFinish={jest.fn()}
+				{...props}
+				value={'foo'}
+			/>,
+		);
+		const input = wrapper.find('input').at(0);
+		input.simulate('change', { target: { value: 'fo' } });
+		expect(wrapper.find(Typeahead).props().items.length).not.toBe(0);
+
+		// when
+		input.simulate('keydown', { which: keycode.codes.enter });
+		input.simulate('blur');
```

Comment:
Remove that, it reset the suggestions too, so you don't really test keydown if you add blur.